src/executor/graph_executor.cc

```diff
@@ -548,12 +548,17 @@ void GraphExecutor::InitCachedOps() {
       vars.resize(std::unique(vars.begin(), vars.end()) - vars.begin());
     };
     dedup(use_vars);
+    for (auto v : use_vars) {
+      if (std::binary_search(mutate_vars.begin(), mutate_vars.end(), v)) {
+        LOG(FATAL) << "var duplication happens for op " << inode.source->attrs.name;
```

Comment:
why not just remove duplication with DeduplicateVarHandles in engine.h. Isn't that how it was usually handled?